src/reason_parser.mly

```diff
@@ -932,6 +933,7 @@ conflicts.
 
 */
 /* Question: Where is the SEMI explicit precedence? */
+%left VARIABLEINFIXOP
```

Comment:
Order matters here. 

putting VARIABLEINFIXOP as first item means this operator has lowest precedence. From what you are doing in the printer and what Haskell uses, I think we need it to be the highest precedence.